### PR TITLE
simplified dye crafting

### PR DESCRIPTION
**Pull request by Napiophelios**
This is a simplified version of dye craft recipes
I don't know how it would affect mods that depend on dye mod
I don't really use any. 

most of the recipes seem legit,
but I am not an expert on actual color mixing.

this is in response to #1491

### CODE REVIEW

**Comment by tenplus1** [comment]
Some of the dye recipe's are a little strange but still better than default dye recipe's :)

If it helps check out my own version of dye mod with realistic dye mixing: https://github.com/tenplus1/dye

**Comment by Napiophelios** [comment]
yeah in my little subgame I am working on,
there are a few more recipes added to allow all dyes to be crafted from other dyes combined,
but I whittled it down to just the ones with no natural sources. 
I think corals could also be possible sources of dye coloring too.
I honestly just don't understand the logic behind the current mixing system.

I will definitely gives yours a test drive later, 
It looks more smarter than mines :P

**Comment by sofar** [comment]
This is not just a simplification - it removes almost all the crafting recipes.

I kind of like the original table, and this removes it. The original table has some nonsense combinations if you ask me, but pruning it and allowing it to have gaps is a much better solution than this.

Not a fan.

**Comment by Napiophelios** [comment]
yeah I didn't really expect many people to be a fan of it :) 

The version I use in my subgame has about 18 recipes altogether
with a means to craft all thru mixing other dyes together.

this PR was just the "barebones" version of it to get all the dyes craftable
without the weird duplicates.

**Comment by paramat** [comment]
I quite like this, however the current code allows a result from every combination.

**Comment by tenplus1** [comment]
The current code supplies a mess of dye recipe's, most of which are quite silly and unrealistic.

**Comment by cx384** [comment]
Why do you want to change the black dye recipe from `group:coal` to `default:coal_lump` ?
https://github.com/minetest/minetest_game/pull/1496/files#diff-a589ed181a1a6a1d1bf2c0a3b98342a6R75

**Comment by Napiophelios** [comment]
the same reason I want to remove "group:stick" from everything...its pointless.

File: mods/dye/init.lua [hunk 1]
```diff
@@ -65,45 +65,66 @@ for _, row in ipairs(dyes) do
...
-	brown      = {"brown",  "brown",  "brown",  "orange",    "brown"},
-	orange     = {"orange", "red",    "orange", "orange"},
-	red        = {"pink",   "magenta","red"},
-	magenta    = {"magenta","magenta"},
-	pink       = {"pink"},
-}
+minetest.register_craft({
+	type = "shapeless",
+	output = "dye:brown 2",
+	recipe = {'dye:blue', 'dye:red', 'dye:yellow'},
```
Comment (paramat) [inline mods/dye/init.lua]: Brown is darkened orange, orange is a mix of red and yellow, how about mixing black, red and yellow?

File: mods/dye/init.lua [hunk 2]
```diff
@@ -65,45 +65,66 @@ for _, row in ipairs(dyes) do
...
+minetest.register_craft({
+	type = "shapeless",
+	output = "dye:grey 2",
+	recipe = {'dye:black', 'dye:white'},
+})
+
+minetest.register_craft({
+	type = "shapeless",
+	output = "dye:grey 2",
+	recipe = {'dye:dark_grey', 'dye:white'},
```
Comment (paramat) [inline mods/dye/init.lua]: I prefer there's 1 recpe for grey, being midway between black and white the other recipe is ideal.

File: mods/dye/init.lua [hunk 3]
```diff
@@ -65,45 +65,66 @@ for _, row in ipairs(dyes) do
...
+})
+
+minetest.register_craft({
+	type = "shapeless",
+	output = "dye:magenta 2",
+	recipe = {'dye:red', 'dye:pink'},
```
Comment (paramat) [inline mods/dye/init.lua]: Pink is just pale red so this is just red plus red. If you search 'magenta colour wheel' you can see it is mainly defined as a mix of red and blue.

**Comment by paramat** [comment]
Added line comments.
Perhaps we can consider making green dye craftable from grass? Since grasses are now everywhere flowers are.

File: mods/dye/init.lua (hunk 3 above)
Comment (Napiophelios) [inline mods/dye/init.lua]: I thought that red and blue would make purple.

Comment (paramat) [inline mods/dye/init.lua]: Well, purple is not used on colour wheels.
Red plus pink makes dark pink.

**Comment by paramat** [comment]
I think your recipes here are much more intuitive than what we have currently so i'm likely to support this PR.

**Comment by Napiophelios** [comment]
I wonder if such changes would conflict with mods like unified dyes etc.
I have never used them so IDK.

**Comment by paramat** [comment]
Yes i'm unsure about possible conflicts or breakages, so this needs more inout.

**Comment by sofar** [comment]
I actually don't like the new PR style, and I think you're missing a ton of combinations that are valid and should have an outcome.

I really do not think that it's possible to go from 105 recipes to 7 without at least explaining for each of the 98 missing ones why they should be removed.

the 105 table is bad (it should have gaps/holes), but the concept and structure are *good*. This new code is just a few randomly picked colors, without any structure.

And I can easily spot missing, crucial, recipes.

:-1:

**Comment by Napiophelios** [comment]
well I am not arguing you are wrong Sofar,
but the point was to reduce 105 recipes down to a rational amount
that's where the "simplified" part kicks in :D

but come on, they aren't randomly selected colors.

Having a table calculate all the possible combinations of the colors listed 
and saying its better because some of the mixtures makes sense
isn't any better or worse than my hacky hodge podge of recipes

**Comment by paramat** [comment]
It might.
The PR recipes are not random or without structure. They provide the dyes not directly craftable in a few intuitive and easy to remember recipes. The current code is so complex that it is difficult to remember how to get a certain dye and impossible to remember all combinations, it also bloats the code with 100 crafting recpes.

**Comment by sofar** [comment]
I posted #1507, which I think is a reasonable pruning of the table, but maintains a bit of structure and explains the recipes listed a bit better.

Please take a look.

**Comment by Napiophelios** [comment]
I think #1507 is more complimentary in "style" to the existing code
Is it okay to close this?

**Comment by paramat** [comment]
Yes. I think i prefer the simplicity of your number of recipes, but the structure of the other PR is better.

**Pull request closed without merging**